src/sys/mod.rs

```diff
@@ -143,6 +143,10 @@ feature! {
 #[allow(missing_docs)]
 pub mod sysinfo;
 
+#[allow(missing_docs)]
+#[cfg(any(bsd, target_os = "linux", not(target_os = "android")))]
```

Comment:
And it is always ok to not support all the platforms, we can add them incrementally

Reply:
Fixed the redundant `android` target. BSD is still not building and I don't really understand why. Something to do with macro? I'm not proficient with them yet.

Reply:
> BSD is still not building and I don't really understand why.

The OpenBSD and NetBSD CI failed because some missing constants in the [libc](https://github.com/rust-lang/libc) crate, which means we need to add them before merging this PR if we want to have `#[cfg(bsd)`s supported.

Let me take a closer look at this and probably give it a fix!